lib/tasks/redirect_legacy_email_signups.rake

```diff
@@ -0,0 +1,31 @@
+desc "Redirect legacy email signups"
+task redirect_legacy_email_signups: :environment do
+  specialist_sectors_info = Topic.where(state: %w[draft published]).where.not(parent_id: nil).map do |ss|
+    { state: ss.state, slug: ss.full_slug }
+  end
+
+  specialist_sectors_info.map do |specialist_sector|
+    base_path = "/topic/#{specialist_sector[:slug]}/email-signup"
+    destination = "/topic/#{specialist_sector[:slug]}"
```

Comment:
Should the destination be "/email-signup?topic=" so they remain sent to the same page as now? 

I can see in discussion on the card it's stated that this isn't possible but I think Ben just got caught with url encoding.

As an example:

```
irb(main):001:0> content_id = SecureRandom.uuid
=> "af079a0c-7333-477b-baf7-2c2efe748b97"
irb(main):002:0> redirect = {
irb(main):003:1*   base_path: "/topic/immigration-operational-guidance/non-compliance-biometric-registration/email-signup",
irb(main):004:1*   document_type: "redirect",
irb(main):005:1*   schema_name: "redirect",
irb(main):006:1*   publishing_app: "collections-publisher",
irb(main):007:1*   update_type: "major",
irb(main):008:1*   redirects: [
irb(main):009:2*     { path: "/topic/immigration-operational-guidance/non-compliance-biometric-registration/email-signup", type: "exact", destination: "/email-signup?topic=#{ERB::Util.url_encode('/topic/immigration-operational-guidance/non-compliance-biometric-registration')}" },
irb(main):010:2*   ],
irb(main):011:1* }
=> {:base_path=>"/topic/immigration-operational-guidance/non-compliance-biometric-registration/email-signup", :document_type=>"redirect", :schema_name=>"redirect", :publishing_app=>"collections-publisher", :update_type=>"major", :redirects=>[{:path=>"/topic/immigration-operational-guidance/non-compliance-biometric-registration/email-signup", :type=>"exact", :destination=>"/email-signup?topic=%2Ftopic%2Fimmigration-operational-guidance%2Fnon-compliance-biometric-registration"}]}
irb(main):012:0> Services.publishing_api.put_content(content_id, redirect)
=> #<GdsApi::Response:0x00007f0036a20c50 @http_response=<RestClient::Response 200 "{\"auth_bypa...">, @web_urls_relative_to=nil>
irb(main):013:0> Services.publishing_api.publish(content_id)
=> #<GdsApi::Response:0x00007f0036a32838 @http_response=<RestClient::Response 200 "{\"content_i...">, @web_urls_relative_to=nil>
```

Reply:
Huh, that's strange - looks like you're right. However, we'd still have to collect the individual topic_ids for these pages, which seems like excessive effort.

We took this approach with Service Manual Publisher, and it has the advantage of following the parent page, if it were ever itself unpublished.

Reply:
I've decided to take Kevin's approach of redirecting to the email-signup path as it seems more useful then redirecting to the parent page. 

I'm not sure what you mean @benthorner when you say 'excessive effort' in retrieving the topic id's?

Reply:
Hmm...I don't think we should do that: if the topic is unpublished in the future, these links will still work. We would actually want them to stop working.

> when you say 'excessive effort' in retrieving the topic id's

Ah, I was getting confused between `topic` and `topic_id`. I should just stop reviewing this PR, sorry 😬. Yes, that would work. However, I still think the above argument is a good one.

Reply:
Either approach you'd have the suffix links still working and returning 301. Is your concern based on whether Email Alert Frontend can handle a topic that no-longer exists? If I remember right it looks up a content item and seems to 404 on non-existent ones e.g. https://www.integration.publishing.service.gov.uk/email-signup?topic=%2Ftopic%2Fnon-existent ?

Reply:
> whether Email Alert Frontend can handle a topic that no-longer exists

I'm seeing redirecting to the parent as simpler, because it's not coupled to the behaviour of another app. It's also the approach we took with service manual signups, so consistent in that respect.

Since the redirect is to the parent, it's not blocking the user. I don't feel too strongly either way, apart from this being a second change in approach, for something that seems pretty arbitrary to me.

I'm happy for you and @1pretz1 to agree on what to do 👍.

Reply:
I've changed tact again 😄 

As consistency is king and these redirects should always be useful if pointing at the parent page as there should always be an end destination that isn't a 404. That okay with you @kevindew?

Reply:
Yup no problem, seems weird to do it that way but don't care strongly enough to argue about it.